zingolib/src/wallet/traits.rs

```diff
@@ -1214,12 +1214,17 @@ where
                 .to_bytes(),
         )?;
 
+        let confirmed_spend = self
+            .spend()
+            .as_ref()
+            .and_then(|(txid, status)| status.get_confirmed_height().map(|height| (txid, height)));
+
         Optional::write(
             &mut writer,
-            self.spent().as_ref(),
+            confirmed_spend,
             |w, (transaction_id, height)| {
                 w.write_all(transaction_id.as_ref())?;
-                w.write_u32::<LittleEndian>(*height)
+                w.write_u32::<LittleEndian>(height.into())
```

Comment:
We need to test make sure this saves and loads correctly.

Reply:
PASS [   0.017s] libtonode-tests::concrete slow::load_wallet_from_v26_dat_file
        PASS [   0.026s] libtonode-tests::concrete fast::reload_wallet_from_buffer
        PASS [   0.047s] libtonode-tests::concrete fast::load_and_parse_different_wallet_versions
        SLOW [> 60.000s] libtonode-tests::concrete slow::load_old_wallet_at_reorged_height
        PASS [  64.242s] libtonode-tests::concrete slow::load_old_wallet_at_reorged_height